x-pack/plugins/siem/server/utils/build_validation/route_validation.test.ts

```diff
@@ -9,9 +9,33 @@ import * as rt from 'io-ts';
 import { RouteValidationResultFactory } from '../../../../../../src/core/server/http';
 
 describe('buildRouteValidation', () => {
-  const schema = rt.type({
-    ids: rt.array(rt.string),
-  });
+  const schema = rt.exact(
+    rt.type({
+      ids: rt.array(rt.string),
+    })
+  );
+  type Schema = rt.TypeOf<typeof schema>;
```

Comment:
NOTE: I use `.exact` in order for the invalid keys to begin showing up with the exact check. Any schema's which miss using the `.exact` is going to allow the front end to push additional keys which will end up in the backend and the exact checker will not find those.

If you look below at the recursive structure you will notice that I show an example where you _must_ use it for each nested level of an io-ts schema to get exact checks working correctly.

Reply:
This is a super important note. I wonder if we could enforce this with a custom linter rule?

Reply:
I don't know if that's needed or possible easily. It is a super important note for users of io-ts that use it for route validation, I will give it that! :-)